packages/shared-core/src/tests/filters.spec.ts

```diff
@@ -0,0 +1,156 @@
+import { buildQuery } from "../filters"
+import {
+  BasicOperator,
+  EmptyFilterOption,
+  FieldType,
+  UILogicalOperator,
+  UISearchFilter,
+} from "@budibase/types"
+
+describe("filter to query conversion", () => {
+  it("handles a filter with 1 group", () => {
```

Comment:
Thank you for doing this, I started it a couple of times in different PRs but never ended up merging it!